### ISSUE TITLE
Status 500 when db.login() on IrishCouch

### ISSUE THREAD

**Issue by LeoLeBras** [nolanlawson/pouchdb-authentication#56]
I am getting an error when I try to login using the authentication plugin : pouchdb-authentification v0.4.1 :
```PouchDB error: the remote database does not seem to have CORS enabled. To fix this, please enable CORS```.

But I have CORS set up on my IrisCouch databse : 

| option           |  value                                                    |
|------------------|------------------------------------------------------|
| cors credentials | true                                                 |
| headers          | accept, authorization, content-type, origin, referer |
| methods          | GET, PUT, POST, HEAD, DELETE                         |
| origins          | *                                                    |

This code works (IrishCouch + local CouchDB) :
```
fetch(`https://mydatabse.iriscouch.com_session`, {
    method: 'post',
    headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
    },
    body: JSON.stringify({
        name: username,
        password: password
    })
}).then(response => {
   console.log(response);
}).
```

But not this one with pouchdb-authentification on IrishCouch (this works with a local CouchDB) :
```
import PouchDB from 'pouchdb';
var db = 'https://mydatabase.iriscouch.com/';
PouchDB.plugin(require('pouchdb-authentication'));
var user = new PouchDB(`${db}_users`);
user.login(username, password).then(response => {
   console.log(response);
}).catch(error => {
   console.log(error); 
});
```
Return: ```{status: 500, name: "unknown_error", message: "Database encountered an unknown error Unknown error!  Did you remember to enable CORS?", error: true}```

**UPDATE 3:00 pm :** this works only with Firefox : 
Return: ```Object { ok: true, name: "***", roles: Array[1] }```

Via curl this works : ```curl -X POST -i https://mydatabse.iriscouch.com/_session -d name=***@**.com -d password=****```, return: ```{"ok":true,"name":"**@**.com","roles":["**"]}```

So, it's not an error on my database I think. Do you know how to fix that ?

Thanks.

Leo

**Comment by LeoLeBras** [comment]
**UPDATE 22 sept 12:00 am :**
Safari iPad : ok :+1: 
Chrome Windows : 1 out of 30 times it works :-1: 
Firefox : ok :+1: 
IE 11: login() ok :+1:  ```{name: "....@......", ok: true, roles: Array[1]}``` **but** getUser() error :-1: ```{"status":404,"name":"not_found","message":"missing"}```

**Comment by LeoLeBras** [comment]
@nolanlawson I can give you the access codes to my database if you want to correct that.

**Comment by nolanlawson** [comment]
I think you can just ignore the error message; that is a bogus message accidentally introduced in PouchDB 4.0.1 and fixed in 4.0.3.

Let me know if that helps!
